src/runtime/methodwrapper.cs

```diff
@@ -27,9 +28,29 @@ public MethodWrapper(Type type, string name, string funcType = null)
             ptr = Runtime.PyCFunction_NewEx(mdef, IntPtr.Zero, IntPtr.Zero);
         }
 
+        ~MethodWrapper()
+        {
+            Dispose();
+        }
+
         public IntPtr Call(IntPtr args, IntPtr kw)
         {
             return Runtime.PyCFunction_Call(ptr, args, kw);
         }
+
+        public void Dispose()
+        {
+            if (_disposed)
+            {
+                return;
+            }
+            _disposed = true;
```

Comment:
You are calling `Dispose` from the finalizer which may execute without the GIL, so you probably have to wrap this in a `using (Py.GIL())`

Reply:
`Dispose` will only call from `Shutdown` now, so I ignore it.
It will be modify after finalizer branch.
![default](https://user-images.githubusercontent.com/3406561/45145438-34743380-b1f3-11e8-8314-936ebe2b57df.png)